src/db/collections/outputs/mod.rs

```diff
@@ -277,6 +277,55 @@ impl OutputCollection {
         .await
     }
 
+    /// Stream all [`LedgerOutput`]s that were unspent at a given ledger index.
+    pub async fn get_unspent_output_stream(
+        &self,
+        ledger_index: MilestoneIndex,
+    ) -> Result<impl Stream<Item = Result<LedgerOutput, Error>>, Error> {
+        self.aggregate(
+            vec![
+                doc! { "$match": {
+                    "metadata.spent_metadata.spent.milestone_index": { "$not": { "$lte": ledger_index } }
+                } },
+                doc! { "$project": {
+                    "output_id": "$_id",
+                    "block_id": "$metadata.block_id",
+                    "booked": "$metadata.booked",
+                    "output": "$output",
+                    "rent_structure": "$details.rent_structure",
+                } },
+            ],
+            None,
+        )
+        .await
+    }
+
+    /// Get all ledger updates (i.e. consumed [`Output`]s) for the given milestone.
+    pub async fn get_ledger_updates(&self, ledger_index: MilestoneIndex) -> Result<Vec<(OutputId, Output)>, Error> {
```

Comment:
Is there a reason why we collect into a `Vec`? If we return an iterator, we could save this allocation. We already do that for `get_unspent_output_stream`.

Reply:
I think for this particular data `(OutputId, Output)` it's save to directly collect in to a `HashMap`, which then can be moved.